paint/support/procedural/src/construct_runtime/mod.rs

```diff
@@ -0,0 +1,392 @@
+    let validate_unsigned = decl_validate_unsigned(&name, modules.iter(), &scrate);
+
+    Ok(quote!(
+        #scrate_decl
+
+        #[derive(Clone, Copy, PartialEq, Eq)]
+        #[cfg_attr(feature = "std", derive(Debug))]
+        pub struct #name;
+        impl #scrate::sr_primitives::traits::GetNodeBlockType for #name {
+            type NodeBlock = #node_block;
+        }
+        impl #scrate::sr_primitives::traits::GetRuntimeBlockType for #name {
+            type RuntimeBlock = #block;
+        }
+
+        #outer_event
+
+        #outer_origin
+
+        #all_modules
+
+        #dispatch
+
+        #metadata
+
+        #outer_config
+
+        #inherent
+
+        #validate_unsigned
+    )
+    .into())
+}
+
+fn decl_validate_unsigned<'a>(
+    runtime: &'a Ident,
+    module_declarations: impl Iterator<Item = &'a ModuleDeclaration>,
+    scrate: &'a TokenStream2,
+) -> TokenStream2 {
+    let modules_tokens = module_declarations
+        .filter(|module_declaration| {
+            module_declaration
+                .module_parts()
+                .into_iter()
+                .any(|part| part.name == "ValidateUnsigned")
+        })
+        .map(|module_declaration| &module_declaration.name);
+    quote!(
+        #scrate::impl_outer_validate_unsigned!(
+            impl ValidateUnsigned for #runtime {
+                #( #modules_tokens )*
+            }
+        );
+    )
+}
+
+fn decl_outer_inherent<'a>(
+    block: &'a syn::TypePath,
+    unchecked_extrinsic: &'a syn::TypePath,
+    module_declarations: impl Iterator<Item = &'a ModuleDeclaration>,
+    scrate: &'a TokenStream2,
+) -> TokenStream2 {
+    let modules_tokens = module_declarations.filter_map(|module_declaration| {
+        let maybe_config_part = module_declaration
+            .module_parts()
+            .into_iter()
+            .filter(|part| part.name == "Inherent")
+            .next();
+        maybe_config_part.map(|config_part| {
+            let arg = config_part
+                .args
+                .as_ref()
+                .and_then(|parens| parens.content.inner.iter().next())
+                .unwrap_or(&module_declaration.name);
+            let name = &module_declaration.name;
+            quote!(#name : #arg,)
+        })
+    });
+    quote!(
+        #scrate::impl_outer_inherent!(
+            impl Inherents where Block = #block, UncheckedExtrinsic = #unchecked_extrinsic {
+                #(#modules_tokens)*
+            }
+        );
+    )
+}
+
+fn decl_outer_config<'a>(
+    runtime: &'a Ident,
+    module_declarations: impl Iterator<Item = &'a ModuleDeclaration>,
+    scrate: &'a TokenStream2,
+) -> TokenStream2 {
+    let modules_tokens = module_declarations
+        .filter_map(|module_declaration| {
+            let generics: Vec<_> = module_declaration
+                .module_parts()
+                .into_iter()
+                .filter(|part| part.name == "Config")
```

Comment:
There only can exists one `Config` per line.

Use the proposed: `ModuleDeclaration::find().map()`

Reply:
To me, it looked like the old decl macro used `first one wins` rule for multiple declarations. So if we're gonna disallow the double declarations I'm gonna need to add it to code. In this case, I also need to understand what to do with `Balances::{default, Config}` - is that an error?

Reply:
Yeah, we should disallow multiple. Adding this to the declarative macro would be a huge pain.

`default, Config` should also be an error.